subsys/usb/class/audio/audio.c

```diff
@@ -10,3 +10,952 @@
+{
+	const struct cs_ac_interface_descriptor_header *header;
+	const struct feature_unit_descriptor *fu;
+
+	header = audio_dev_data->header_descr;
+	fu = (struct feature_unit_descriptor *)((u8_t *)header +
+						header->bLength +
+						INPUT_TERMINAL_DESC_SIZE);
+	if (fu->bUnitID == entity->id) {
+		entity->subtype = fu->bDescriptorSubtype;
+		return true;
+	}
+
+	if (header->bInCollection == 2) {
+		fu = (struct feature_unit_descriptor *)((u8_t *)fu +
+			fu->bLength +
+			INPUT_TERMINAL_DESC_SIZE +
+			OUTPUT_TERMINAL_DESC_SIZE);
+		if (fu->bUnitID == entity->id) {
+			entity->subtype = fu->bDescriptorSubtype;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * @brief Helper funciton for getting the audio_dev_data by the entity number.
+ *
+ * This function searches through all audio devices the one with given
+ * entity number and return the audio_dev_data structure for this entity.
+ *
+ * @param [in, out] entity USB Audio entity addressed by the request.
+ *			   .id      [in]  id of searched entity
+ *			   .subtype [out] subtype of entity (if found)
+ *
+ * @return audio_dev_data for given entity, NULL if not found.
+ */
+static struct usb_audio_dev_data_t *get_audio_dev_data_by_entity(
+					struct usb_audio_entity *entity)
+{
+	struct usb_dev_data *dev_data;
+	struct usb_audio_dev_data_t *audio_dev_data;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&usb_audio_data_devlist, dev_data, node) {
+		audio_dev_data = CONTAINER_OF(dev_data,
+					      struct usb_audio_dev_data_t,
+					      common);
+
+		if (is_entity_valid(audio_dev_data, entity)) {
+			return audio_dev_data;
+		}
+	}
+	return NULL;
+}
+
+/**
+ * @brief Helper funciton for checking if particular interface is a part of
+ *	  the audio device.
+ *
+ * This function checks if given interface is a part of given audio device.
+ * If so then true is returned and audio_dev_data is considered correct device
+ * data.
+ *
+ * @param [in] audio_dev_data USB audio device data.
+ * @param [in] interface      USB Audio interface number.
+ *
+ * @return true if interface matched audio_dev_data, false otherwise.
+ */
+static bool is_interface_valid(struct usb_audio_dev_data_t *audio_dev_data,
+			       u8_t interface)
+{
+	const struct cs_ac_interface_descriptor_header *header;
+
+	header = audio_dev_data->header_descr;
+	u8_t desc_iface = 0;
+
+	for (size_t i = 0; i < header->bInCollection; i++) {
+		desc_iface = header->baInterfaceNr[i];
+		if (desc_iface == interface) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * @brief Helper funciton for getting the audio_dev_data by the interface
+ *	  number.
+ *
+ * This function searches through all audio devices the one with given
+ * interface number and returns the audio_dev_data structure for this device.
+ *
+ * @param [in] interface USB Audio interface addressed by the request.
+ *
+ * @return audio_dev_data for given interface, NULL if not found.
+ */
+static struct usb_audio_dev_data_t *get_audio_dev_data_by_interface(
```

Comment:
as this is static, why not use shorter names without prefix like `static struct audio_context *get_dev_data_by_iface`

Reply:
For sure not `get_dev_data_by_iface` as `usb_get_dev_data_by_iface` is used in the usb_descriptor.h and this could be misleading. Will replace with:
`static struct usb_audio_data *get_audio_data_by_iface()`

Reply:
According to previous comment: `get_audio_dev_data_by_iface()`